src/poetry/core/masonry/builder.py

```diff
@@ -21,7 +21,12 @@ def __init__(self, poetry: Poetry) -> None:
             "wheel": WheelBuilder,
         }
 
-    def build(self, fmt: str, executable: str | Path | None = None) -> None:
+    def build(
+        self,
+        fmt: str,
+        output_directory: Path | None = None,
+        executable: str | Path | None = None,
```

Comment:
```suggestion
        *,
        output_directory: Path | None = None,
        executable: str | Path | None = None,
```

Just some quick fly-by feedback: If we are sure that we can add the new parameter between `fmt` and `executable` without breaking anything (I've not checked if that is true.), we should take the chance and make these keyword-only arguments.

Reply:
I would prefer to add the new parameter at the end. Thus we can makes sure we will not break anything. Furthermore I would prefer naming it `target_dir`, just like in the `build()` method in `Builder`.

Reply:
I'm also happy to put the output directory after executible if that would be better?

Reply:
I'm fine with that. It's the safe option. Maybe, we can make at least the new option keyword-only.

Reply:
Recent change places `target_dir` at the end and makes it keyword-only